subprojects/integ-test/src/integTest/groovy/org/gradle/integtests/CacheProjectIntegrationTest.groovy

```diff
@@ -93,6 +93,7 @@ class CacheProjectIntegrationTest extends AbstractIntegrationTest {
         classFile.assertHasChangedSince(classFileSnapshot)
         classFileSnapshot = classFile.snapshot()
 
+        executer.expectDeprecationWarning().requireGradleDistribution()
```

Comment:
❌ Why was `requireGradleDistribution` added here?

Reply:
Without `requireGradleDistribution`, the warning will be printed at the same JVM with the test code, thus we can't catch it.

Reply:
I don't think that's the case, there are plenty of tests using `expectDeprecationWarning`, but not `requireGradleDistribution`.

Reply:
Things are a bit different here. In this case, the warning message is outputted during parameter parsing, which happens before we can catch the build message: https://github.com/gradle/gradle/blob/master/subprojects/internal-integ-testing/src/main/groovy/org/gradle/integtests/fixtures/executer/InProcessGradleExecuter.java#L267

Reply:
In other cases, the outputted deprecation message would be catched by this: https://github.com/gradle/gradle/blob/master/subprojects/internal-integ-testing/src/main/groovy/org/gradle/integtests/fixtures/executer/InProcessGradleExecuter.java#L279

That's why other build operations don't need isolated distribution.

Reply:
Ah, that's what I was missing. Thanks for the explanation!